src/include/btmem.h

```diff
@@ -410,10 +409,24 @@ struct __wt_page {
 				uint32_t entries;
 				WT_REF	**index;
 			} * volatile __index;	/* Collated children */
+
+			/*
+			 * When splitting to deepen the tree, track the number
+			 * of entries in the newly created parent, and how many
+			 * subsequent splits follow the initial set of entries.
+			 * If future splits into the page are generally after
+			 * the initial set of items, perform future deepening
+			 * splits in this page to optimize for an append-style
+			 * workload.
+			 */
+			uint32_t deepen_split_append;
+			uint32_t deepen_split_last;
```

Comment:
To recap, I'm tracking the number of items created by the last deepen-split, and the number of subsequent splits into the page that were after that initial number of items.

When we do a deepen-split, if 80% of the subsequent splits into the page were after the items created in the last deepen-split, we ignore the items created in the last deepen-split.

Obviously, this only handles append-style workloads.

I think the next step would be to timestamp each new entry as it's split into the page, and then when we deepen the tree, we'd review all of the entries and push down new entries (but, of course, we'd have to detect ranges of new entries, pushing down single new entries isn't useful). That's quite a bit more work, and the timestamp would probably have to go into the WT_REF (which shouldn't grow), so there's some trickiness involved.

Anyway, I don't like this solution/idea much, can anyone think of a better one?